tuf/repository_tool.py

```diff
@@ -1777,7 +1778,64 @@ def __call__(self, rolename):
 
 
 
-  def add_delegated_role(self, rolename, targets_object):
+
+  def load_signing_key_succinct_delegations(self, key):
+    """
+    <Purpose>
+      Load the role key for all succinct delegations. All succinct delegations
+      use the same role key
+
+    <Arguments>
+      key:
+        The role key for all succinct delegations associated with this role,
+        conformant to 'securesystemslib.formats.ANYKEY_SCHEMA'.
+        It must contain the private key, so that role signatures may be
+        generated when writeall() or write() is eventually called to generate
+        valid metadata files.
+
+    <Exceptions>
+      securesystemslib.exceptions.FormatError, if 'key' is improperly formatted.
+
+      securesystemslib.exceptions.Error, if the private key is not found in 'key'.
+
+    <Side Effects>
+      Updates the keydb and roledb entries for all succinct delegations
+
+    <Returns>
+      None.
+
+    """
+    # Does 'key' have the correct format?
+    # Ensure the arguments have the appropriate number of objects and object
+    # types, and that all dict keys are properly named.  Raise
+    # 'securesystemslib.exceptions.FormatError' if any are improperly formatted.
+    securesystemslib.formats.ANYKEY_SCHEMA.check_match(key)
+
+    # Ensure the private portion of the key is available, otherwise signatures
+    # cannot be generated when the metadata file is written to disk.
+    if 'private' not in key['keyval'] or not len(key['keyval']['private']):
```

Comment:
I like dict's `get` here, it's slightly shorter (and therefore more readable) and avoids querying the dict twice.

```suggestion
    if not key['keyval'].get('private'):
```